network-api/networkapi/campaign/views.py

```diff
@@ -30,6 +30,21 @@
 logger = logging.getLogger(__name__)
 
 
+@api_view(['POST'])
+@parser_classes((JSONParser,))
+@permission_classes((permissions.AllowAny,))
+def signup_submission_view(request, pk):
```

Comment:
this is essentially a duplication of the petition code, but I wanted to keep the route view "its own thing" instead of making both signup and petition go to the same def and have its code unravel the actual url things came in on, which would make things far harder to read.